src/utils/socialPicture/index.ts

```diff
@@ -148,14 +149,16 @@ export const cacheSocialPictureInCDN = async (
   }
   const fileName = getNFTFilenameInCDN(socialPic);
   const fileNameWithOverlay = getNFTFilenameInCDN(socialPic, domain.name);
+  const fileNameJpeg = getNFTFilenameInCDN(socialPic)?.replace('.svg', '.jpg'); // dirty hack, pass param instead
+  const fileNameWithOverlayJpeg = getNFTFilenameInCDN(socialPic, domain.name)?.replace('.svg', '.jpg');
   const bucketName = env.CLOUD_STORAGE.CLIENT_ASSETS.BUCKET_ID;
   const bucket = storage.bucket(bucketName);
 
   const [fileExists] = await bucket.file(fileName).exists();
-  const [fileWithOverlayExists] = await bucket
-    .file(fileNameWithOverlay)
-    .exists();
-  if (!fileExists || !fileWithOverlayExists) {
+  const [fileWithOverlayExists] = await bucket.file(fileNameWithOverlay).exists();
+  const [fileJpegExists] = await bucket.file(fileNameJpeg).exists();
+  const [fileWithOverlayJpegExists] = await bucket.file(fileNameWithOverlayJpeg).exists();
```

Comment:
Consider parallelising requests:
```suggestion
  const [fileExists , fileWithOverlayExists, fileJpegExists, fileWithOverlayJpegExists] = await Promise.all([
    bucket.file(fileName).exists(),
    bucket.file(fileNameWithOverlay).exists(),
    bucket.file(fileNameJpeg).exists(),
    bucket.file(fileNameWithOverlayJpeg).exists()
  ]);
```

Reply:
Alternatively, have you tried this?
```ts
const files = await bucket.getFiles({
  prefix: fileName
});
```

You can extend `getNFTFilenameInCDN` to support an argument to return a file name without an extension.

Reply:
Let me try that..